File: day01/day1.cpp


#include <iostream>
#include <fstream>
#include <string>

using namespace std;

int calc_fuel(int mass) {
   int sum = 0;
   while(mass > 0) {
      int fuel = (mass / 3) - 2;
      if (fuel > 0) sum += fuel;
      mass = fuel;
   }
   return sum;
}

int main() {
   ifstream input("input.txt");

   int n;
   int sum_1 = 0;
   int sum_2 = 0;
   while (input >> n) {
      sum_1 += (n / 3) - 2;
      sum_2 += calc_fuel(n);
   }

   cout << "Part 1: " << sum_1 << "\n";

   cout << "Part 2: " << sum_2 << "\n";
}

File: day01/day1.py
def fuel_for(weight):
  total = 0
  while weight > 0:
    fuel = (weight // 3) - 2
    if (fuel > 0):
      total += fuel
    weight = fuel
  return total

with open("input.txt") as f:
  weights = [int(x) for x in f.readlines()]
  
  part1 = sum([((x // 3) - 2) for x in weights])
  print ("Part 1: ", part1)

  part2 = sum([fuel_for(x) for x in weights])
  print ("Part 2: ", part2)
